App/Containers/Root/RootScreen.js

```diff
@@ -1,37 +1,57 @@
 import React, { Component } from 'react'
 import NavigationService from 'App/Services/NavigationService'
 import AppNavigator from 'App/Navigators/AppNavigator'
-import { View } from 'react-native'
+import { View, Text } from 'react-native'
 import { connect } from 'react-redux'
 import StartupActions from 'App/Stores/Startup/Actions'
 import { PropTypes } from 'prop-types'
 import { Helpers } from 'App/Theme'
+import LoginScreen from '../LoginScreen/LoginScreen'
 
 class RootScreen extends Component {
   componentDidMount() {
-    // Run the startup saga when the application is starting
-    this.props.startup()
+    // Run the startup saga only if userId is available
+    if (this.props.userId !== null)
+       this.props.startup()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.userId !== prevProps.userId) {
+      // Start the app after user entered id
+        this.props.startup()
+    }
   }
 
   render() {
     return (
       <View style={Helpers.fill}>
-        <AppNavigator
+      {
+        this.props.userId === null ?
```

Comment:
this could be just `this.props.userId ? ...`

Reply:
walla